src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -4877,6 +4877,30 @@ @preservewhitespace someVariable
         Assert.Collection(generated.Diagnostics, d => { Assert.Equal("RZ1038", d.Id); });
     }
 
+    [IntegrationTestFact, WorkItem("https://github.com/dotnet/razor/issues/7169")]
+    public void InheritsDirective_NullableReferenceType()
+    {
+        // Arrange
+        AdditionalSyntaxTrees.Add(Parse("""
+            namespace Test;
+
+            public class BaseComponent<T> : Microsoft.AspNetCore.Components.ComponentBase { }
```

Comment:
Let's directly observe the `T` from the razor component so we can see nullable warnings getting reported through the whole pipeline.